src/lib/coda_networking/coda_networking.ml

```diff
@@ -68,8 +68,19 @@ module Rpcs = struct
         let caller_model_of_response = Fn.id
       end
 
+      module T_bin_io = Perf_histograms.Rpc.Plain.Decorate_bin_io (struct
+        let name = name
+
+        include T
+      end)
+
       include T
-      include Register (T)
+      include T_bin_io
```

Comment:
There are 2 places with `T` followed by `T_bin_io`, and I think you're relying that the functions in `T_bin_io` shadow those in `T`. So that's a little fragile.

It might be better if the `Decorate...` functor returned not only the shadowed functions, but also the unshadowed ones in `T`. Then, you could just put `T_bin_io`, and not have to rely on the order of the `include`s.